pagefaultgames/pokerogue
# [Balance] Move-based Tyrogue Evolutions

## What are the changes the user will see?
Tyrogue now learns 3 new moves at Level 10:
- Low Sweep
- Mach Punch
- Rapid Spin

At Level 20, Tyrogue now evolves into Hitmonlee, Hitmonchan, or Hitmontop depending on which move it knows:
- Low Sweep -> Hitmonlee
- Mach Punch -> Hitmonchan
- Rapid Spin -> Hitmontop

Tyrogue will only evolve if it knows at least one of these moves. If Tyrogue knows multiple of these moves, it will evolve based on the qualifying move that first occurs in its moveset.

## Why am I making these changes?
Requested by Balance Team. Tyrogue's current evolution method doesn't translate well into PokeRogue's mechanics, especially since IVs cannot be controlled on starter select. This change makes it so all 3 Hitmon-s can reliably be built into when selecting Tyrogue as a starter.

## What are the changes from a developer perspective?
- `data/balance/pokemon-level-moves`: Tyrogue now learns Low Sweep, Mach Punch, and Rapid Spin at level 10.
- `data/balance/pokemon-evolutions`: Tyrogue's evolution conditions are now based on the following process:
  1. Find the first qualifying move (Low Sweep, Mach Punch, or Rapid Spin) in Tyrogue's moveset.
  2. If a qualifying move is found, evolve to that move's respective Pokemon (see the user-facing changes section).
  3. Otherwise, do not evolve.

### Screenshots/Videos

<details>
  <summary> Evolving into Hitmonlee </summary>

https://github.com/user-attachments/assets/4d4a902e-0a30-47c6-9922-54d703b96442

</details>


<details>
  <summary> Evolving into Hitmonchan </summary>

https://github.com/user-attachments/assets/b35077a9-5396-4833-a312-6b1f6547085f

</details>

<details>
  <summary> Evolving into Hitmontop </summary>

https://github.com/user-attachments/assets/5ecbaaeb-ef48-4e59-8043-b9171435a919

</details>

## How to test the changes?

1. Start a game with Tyrogue. For easier testing, you might want to use these overrides:
```ts
const overrides = {
  OPP_MOVESET_OVERRIDE: Moves.SPLASH,
  ITEM_REWARD_OVERRIDE: [{name: "RARE_CANDY"}]
} satisfies Partial<InstanceType<typeof DefaultOverrides>>;
```
2. Have Tyrogue learn Low Sweep, Mach Punch, and/or Rapid Spin when it gets the opportunity.
3. When Tyrogue reaches level 20, it should evolve based on whatever evolve move is first in its moveset.

**Note:** Be careful about using `MOVESET_OVERRIDE` when testing this if you decide to do so. The game will interpret the Tyrogue's moveset only in whatever order you set them in the override (even if you swap them in summary).

## Checklist
